calico_node/startup/startup_test.go

```diff
@@ -52,3 +59,173 @@ var _ = Describe("Non-etcd related tests", func() {
 		})
 	})
 })
+
+type EnvItem struct {
+	key string
+	value string
+}
+
+var _ = Describe("Etcd required tests", func() {
+	changedEnvVars := []string{"CALICO_IPV4POOL_CIDR", "CALICO_IPV6POOL_CIDR", "NO_DEFAULT_POOLS"}
+
+	BeforeEach(func() {
+		for _, envName := range changedEnvVars {
+			os.Unsetenv(envName)
+		}
+	})
+	AfterEach(func() {
+		for _, envName := range changedEnvVars {
+			os.Unsetenv(envName)
+		}
+	})
+
+	DescribeTable("Test IP pool env variables",
+		func(envList []EnvItem, expectPools bool, expectedIPv4 string, expectedIPv6 string) {
+
+			// Erase etcd clean.
+			testutils.CleanEtcd()
+
+			// Create a new client.
+			c, err := testutils.NewClient("")
+			if err != nil {
+				log.Println("Error creating client:", err)
+			}
+
+			// Set the env variables specified.
+			for _, env := range envList {
+				os.Setenv(env.key, env.value)
+			}
+			poolList, err := c.IPPools().List(api.IPPoolMetadata{})
+			Expect(poolList.Items).To(BeEmpty())
+
+			// Run the UUT.
```

Comment:
My intention is Unit Under Test.  Is there a name that the Tigera crew would use there?